docs/docs/developers/tutorials/codealong/first_fees.md

```diff
@@ -0,0 +1,404 @@
+
+**Create the keys and account in the PXE**
+
+Lets first create the account but only register the instance in the PXE. Its address will be calculated, and in a later step we can deploy it to the network.
+
+```bash
+aztec-wallet create-account --register-only -a main
+```
+
+The `-a main` sets this new account's alias to, "main". Use `aztec-wallet get-alias` (with no params) to see all aliases.
+
+The equivalent using aztec.js - create a random account locally (we will deploy it in the next step):
+
+```javascript
+import { Fr } from "@aztec/aztec.js";
+import { getSchnorrAccount } from '@aztec/accounts/schnorr';
+  //... building upon the previous section
+  let secretKey = Fr.random(); let salt = Fr.random();
+  let schnorrAccount = await getSchnorrAccount(pxe, secretKey, deriveSigningKey(secretKey), salt);
+```
+
+Your PXE now has keys for an account that can be deployed on Aztec network(s).
+
+## Payment option overview
+
+To get a quick overview of the payment options currently supported...
+
+**Sponsored Fee Paying Contract (Sponsored FPC)**
+A small amount of funds from a limited faucet-like contract. Useful for bootstraping first tx(s), eg deloying an account
+
+**Fee juice**
+- Claimed from the fee asset bridged from L1
+- From a funded account sending a transaction (default)
+- From a different funded account to deploy an account, (not possible with other tx types, only deployment)
+
+**General Fee Paying Contracts**
+Pay in one asset (publicly or privately) for the FPC to (publicly) pay for the transaction.
+Great for privacy when an account is paying privately, and interacting with a private contract function.
+
+Remember, we will go into these in the following sections.
+
+## Paying for an account deployment transaction
+
+To make transactions on the network, your account contract will need to specify a payment method of the enshrined asset, "Fee Juice".
+
+### Sponsored FPC
+
+To bootstrap first use, a sponsored fee paying contract (the canonical sponsored FPC) can be used to deploy your first account.
+
+**TODO: as snippet**
+A fee paying contract (FPC) effectively implements fee abstraction. It is a contract that pays for transactions of other accounts, when its own custom criteria is met.
+In the case of the canonical sponsored FPC, the only criteria is an upper bound on how much it sponsors an account's transactions. This will be enough to at least deploy an account.
+
+The PXE can be queried for the canonical sponsored FPC address, and then specified as the payment method.
+
+Via the CLI:
+
+The alias set earlier can be confirmed using: `aztec-wallet get-alias accounts:main`, this is specified here in `--from main`.
+
+```bash
+SPONSORED_FPC_ADDRESS=<address>
+aztec-wallet register-contract SPONSORED_FPC_ADDRESS SponsoredFPC --from main # Need to specify account that wishes to register the contract
+aztec-wallet deploy-account --from main --payment method=fpc-sponsored,fpc=contracts:sponsoredfpc
```

Comment:
```suggestion
aztec-wallet deploy-account --from main --payment method=fpc-sponsored,fpc=$SPONSORED_FPC_ADDRESS
```

Reply:
This was aliased in the previous command, but I'll remove the alias to use the env var.